File: server/src/main/java/com/skywalker/backend/service/repo/DoctorService.java
package com.skywalker.backend.service.repo;

import com.skywalker.backend.model.Doctor;
import com.skywalker.backend.dto.DoctorDTO;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface DoctorService {

    Doctor getDoctorById(Long id);

    List<Doctor> getAllDoctors();

    Doctor createDoctor(DoctorDTO dto);

    Doctor updateDoctor(Long id, DoctorDTO dto);

    void deleteDoctor(Long id);
}

File: client/src/services/doctorService.js
import api from './api';

export const doctorService = {
  getAllDoctors: async () => {
    return await api.get('/api/doctors');
  },

  getDoctorById: async (id) => {
    return await api.get(`/api/doctors/${id}`);
  },

  getDoctorsBySpecialization: async (specialization) => {
    return await api.get(`/api/doctors/specialization/${specialization}`);
  },

  updateDoctor: async (id, doctorData) => {
    return await api.put(`/api/doctors/${id}`, doctorData);
  },

  deleteDoctor: async (id) => {
    return await api.delete(`/api/doctors/${id}`);
  },
};
